plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -0,0 +1,337 @@
+								event.payload.userProfile, // sets context.userProfile to the payload of the event
+						} ),
+					],
+				},
+				USER_PROFILE_SKIPPED: {
+					target: 'preBusinessInfo',
+					actions: [
+						assign( {
+							userProfile: ( context, event: UserProfileEvent ) =>
+								event.payload.userProfile, // assign context.userProfile to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 40,
+				component: UserProfile,
+			},
+		},
+		preBusinessInfo: {
+			always: [
+				// immediately transition to businessInfo without any events as long as geolocation parallel has completed
+				{
+					target: 'businessInfo',
+					cond: () => true, // TODO: use a custom function to check on the parallel state using meta when we implement that. https://xstate.js.org/docs/guides/guards.html#guards-condition-functions
+				},
+			],
+			meta: {
+				progress: 50,
+			},
+		},
+		businessInfo: {
+			on: {
+				BUSINESS_INFO_COMPLETED: {
+					target: 'preExtensions',
+					actions: [
+						assign( {
+							businessInfo: (
+								_context,
+								event: BusinessInfoEvent
+							) => event.payload.businessInfo, // assign context.businessInfo to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 60,
+				component: BusinessInfo,
+			},
+		},
+		skipFlowBusinessLocation: {
+			on: {
+				BUSINESS_LOCATION_COMPLETED: {
+					target: 'settingUpStore',
+					actions: [
+						assign( {
+							businessInfo: (
+								_context,
+								event: BusinessLocationEvent
+							) => event.payload.businessInfo, // assign context.businessInfo to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 80,
+				component: BusinessLocation,
+			},
+		},
+		preExtensions: {
+			always: [
+				// immediately transition to extensions without any events as long as extensions fetching parallel has completed
+				{
+					target: 'extensions',
+					cond: () => true, // TODO: use a custom function to check on the parallel state using meta when we implement that. https://xstate.js.org/docs/guides/guards.html#guards-condition-functions
+				},
+			],
+			// add exit action to filter the extensions using a custom function here and assign it to context.extensionsAvailable
+			exit: assign( {
+				extensionsAvailable: ( context ) => {
+					return context.extensionsAvailable.filter( () => true );
+				}, // TODO : define an extensible filter function here
+			} ),
+			meta: {
+				progress: 70,
+			},
+		},
+		extensions: {
+			on: {
+				EXTENSIONS_COMPLETED: {
+					target: 'settingUpStore',
+				},
+			},
+			meta: {
+				progress: 80,
+				component: Extensions,
+			},
+		},
+		settingUpStore: {},
+	},
+} );
+
+const CoreProfilerController = ( {} ) => {
+	const [ state, send ] = useMachine(
+		coreProfilerStateMachineDefinition.withConfig( {
+			services: {
+				initializeProfiler: () => ( sendBack ) => {
+					const interval = setInterval( () => {
```

Comment:
Curious why we need `setInterval` here? Will this be replaced by something later?

Reply:
Yep placeholder just to make the init step appear, after the tasks to implement the extensions fetching and the geolocation, we can remove this